### PR TITLE
[RFC] Fix SETTINGS parse after FORMAT

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix SETTINGS parse after FORMAT

P.S. did not mark as bugfix to avoid red label and because **I'm not sure about this change per se** (but at least it will run tests and perf tests with the settings after FORMAT interpreted correctly)

<details>

HEADs:
- 8ec9c5562472abcab9fde913636e2a35b7977134
- f38898594c8ff5c948b97b902dc9134a1b24cb07

</details>

### CODE REVIEW

File: src/Parsers/ParserQueryWithOutput.cpp [hunk 1]
```diff
@@ -107,6 +110,14 @@ bool ParserQueryWithOutput::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
         if (!parser_settings.parse(pos, query_with_output.settings_ast, expected))
             return false;
         query_with_output.children.push_back(query_with_output.settings_ast);
+
+        // SETTINGS after FORMAT is not parsed by the SELECT parser (ParserSelectQuery)
+        // Pass them manually, to apply in InterpreterSelectQuery::initSettings()
+        if (query->as<ASTSelectWithUnionQuery>())
+        {
+            QueryWithOutputSettingsPushDownVisitor::Data data{query_with_output.settings_ast};
+            QueryWithOutputSettingsPushDownVisitor(data).visit(query);
```
Comment (azat) [inline src/Parsers/ParserQueryWithOutput.cpp:119]: Not sure about a visitor for this, but looks a little bit saner

Comment (azat) [reply src/Parsers/ParserQueryWithOutput.cpp:119]: @alexey-milovidov can you take a look at this please? worth it or not?

**Review by alexey-milovidov — APPROVED** [review]
Totally Ok.

**Comment by azat** [comment]
@alexey-milovidov great, worth changing changelog category to bug fix?

**Comment by alexey-milovidov** [comment]
Perf test: #12459

**Pull request merged**